src/mod/endpoints/mod_sofia/sofia.c

```diff
@@ -9527,6 +9529,123 @@ static switch_status_t sofia_process_proxy_refer(switch_core_session_t *session,
 	return SWITCH_STATUS_FALSE;
 }
 
+static switch_bool_t has_valid_image_media_type(const char *r_sdp)
+{
+	switch_bool_t ret = SWITCH_FALSE;
+	sdp_parser_t *parser = NULL;
+	sdp_session_t *sdp;
+
+	if ((parser = sdp_parse(NULL, r_sdp, (int) strlen(r_sdp), 0))) {
+		if ((sdp = sdp_session(parser))) {
+			sdp_media_t *m;
+			for (m = sdp->sdp_media; m; m = m->m_next) {
+				if (m->m_type == sdp_media_image && m->m_port) {
+					sdp_attribute_t *attr;
+					ret = SWITCH_TRUE;
+					for (attr = m->m_attributes; attr; attr = attr->a_next) {
+						if (!strcasecmp(attr->a_name, "recvonly")) {
+							ret = SWITCH_FALSE;
+							break;
+						}
+					}
+                }
+			}
+		}
+	}
+	if (parser) {
+		sdp_parser_free(parser);
+	}
+	return ret;
+}
+
+void sofia_handle_sip_i_update(nua_t *nua, sofia_profile_t *profile, nua_handle_t *nh, switch_core_session_t *session, sip_t const *sip,
+								sofia_dispatch_event_t *de, tagi_t tags[])
+{
+	const char *r_sdp = NULL;
+	const char *session_id_header = NULL;
+	private_object_t *tech_pvt = NULL;
+	switch_channel_t *channel = NULL;
+	switch_bool_t has_valid_sdp = SWITCH_TRUE;
+
+	if (session && sip && sip->sip_payload && sip->sip_payload->pl_data) {
+		tech_pvt = switch_core_session_get_private(session);
+		r_sdp = sip->sip_payload->pl_data;
+		if (!strcmp(tech_pvt->mparams.remote_sdp_str, r_sdp)) {
+			switch_log_printf(SWITCH_CHANNEL_SESSION_LOG(session), SWITCH_LOG_DEBUG, "Duplicate SDP\n%s\n", r_sdp);
+			has_valid_sdp = SWITCH_FALSE;
```

Comment:
This should be TRUE, we still wanted to respond the last SDP answer from their previous SDP offer

Reply:
ok